src/skill.cpp

```diff
@@ -160,23 +148,27 @@ bool SkillLevel::isRusting() const
 
 bool SkillLevel::rust( bool charged_bio_mem )
 {
-    if (_level > 0 && calendar::turn > _lastPracticed &&
-        (calendar::turn - _lastPracticed) % rustRate(_level) == 0) {
-        if (charged_bio_mem) {
-            return one_in(5);
-        }
-        _exercise -= _level;
-
-        if (_exercise < 0) {
-            if (OPTIONS["SKILL_RUST"] == "vanilla" || OPTIONS["SKILL_RUST"] == "int") {
-                _exercise = (100 * _level) - 1;
-                --_level;
-            } else {
-                _exercise = 0;
-            }
+    calendar const delta = calendar::turn - _lastPracticed;
+    if (_level <= 0 || delta <= 0 || delta % rustRate(_level)) {
+        return false;
+    }
+
+    if (charged_bio_mem) {
+        return one_in(5);
+    }
+
+    _exercise -= _level;
+    auto const &rust_type = OPTIONS["SKILL_RUST"];
+    if (_exercise < 0) {
+        if (rust_type == "vanilla" || rust_type == "int") {
+            _exercise = (100 * _level) - 1;
+            --_level;
+        } else {
+            _exercise = 0;
         }
     }
-    return false;
+
+    return true;
```

Comment:
:frowning: Not sure how false became true, but I'll make a PR if one hasn't already been made; thanks @BevapDin .